### PR TITLE
Introduction to recursion and tracing of recursion for factorial function

### PR DESCRIPTION
**Pull request by SanyaGera**
## Related Issue
-# issue number #1320 

## Proposed Changes
- Made changes in the README file. I added the following topics:
    -Addition of table of contents and navigation links
    -Addition of definition and syntax of recursion
    -Addition of advantages of recursion
    -Addition of disadvantages of recursion
    -Addition of tracing of recursion.
- Added an Introduction to Recursion CPP file containing a basic program on recursion for its explanation.
- Added a CPP file for Factorial function using recursion.
   

## Additional Info
- No additional Info

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
Code:
image
image

Output:
image
image

### CODE REVIEW

File: R-Recursion/README.md [hunk 1]
```diff
@@ -1,9 +1,79 @@
...
+  - [Recursion](#recursion)
+    - [Definition and Syntax](#definition-and-syntax)
+    - [Advantages of recursion](#advantages-of-recursion)
+    - [Disadvantages of recursion](#disadvantages-of-recursion)
+    - [Difference between iterative approach and recursive approach](#difference-between-iterative-approach-and-recursive-approach)
+    - [Tracing of a recursive function](#tracing-of-a-recursive-function)
+    - [Recursion using Stack](#recursion-using-stack)
+    - [Time Complexity in Recursion](#time-complexity-in-recursion)
+    - [Static and Global Variables in Recursion](#static-and-global-variables-in-recursion)
+    -
```
Comment (Mrjoy832) [inline R-Recursion/README.md:12]: remove

File: R-Recursion/README.md [hunk 2]
```diff
@@ -1,9 +1,79 @@
-# [Inversion Count](https://github.com/Lakhankumawat/LearnCPP/blob/main/Day-100(Algorithms)/R-Recursion/Inversion_Count.cpp) 
+# Table Of Content
+
+  - [Recursion](#recursion)
```
Comment (Mrjoy832) [inline R-Recursion/README.md:3]: image
- spelling error
- Add a state Space tree diagram explaining Tracing of Recusrion

**Review by Mrjoy832 — CHANGES_REQUESTED** [review]

File: R-Recursion/Introduction-to-Recursion.cpp [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#include<iostream>
+using namespace std;
```
Comment (Mrjoy832) [inline R-Recursion/Introduction-to-Recursion.cpp:2]: remove this only keep factorial and show tracing of factorial in readme ......change the PR title accordingly

**Review by Mrjoy832 — CHANGES_REQUESTED** [review]

File: R-Recursion/README.md [hunk 4]
```diff
@@ -1,9 +1,76 @@
...
+    - [Definition and Syntax](#definition-and-syntax)
+    - [Advantages of recursion](#advantages-of-recursion)
+    - [Disadvantages of recursion](#disadvantages-of-recursion)
+    - [Difference between iterative approach and recursive approach](#difference-between-iterative-approach-and-recursive-approach)
+    - [Tracing of a recursive function](#tracing-of-a-recursive-function)
```
Comment (Mrjoy832) [inline R-Recursion/README.md:8]: on this point add tracing of factorial with state space tree

**Review by Mrjoy832 — CHANGES_REQUESTED** [review]

**Review by Mrjoy832 — APPROVED** [review]

**Comment by Mrjoy832** [comment]
@Lakhankumawat , Kindly review this one

**Comment by Lakhankumawat** [comment]
Recursion folder is changing too much these days , i've told a contributor to organise it , i'll see this pr after it is done .
Also remove the underscore in file name 
File name must be camel cased without any space

There are too many issues in this Pr 
Links in README are not working
there are two code files and the contribution is too basic i.e. factorial using recursion .

**Comment by Mrjoy832** [comment]
@SanyaGera resolve conflicts

**Comment by SanyaGera** [comment]
Hello @Lakhankumawat @Mrjoy832 , I understand that there are several issues and pull requests being made on recursion everyday. That is why, I created this PR which organizes the README file into topics and subtopics. 
Also, I checked it again and all the links in README are working on my end, if there is some error please let me know.
And the purpose for including a basic file factorial-using-recursion is that I wanted to show tracing of recursion using a basic code and this advise was given to me by @Mrjoy832 sir. 
I hope you will take my PR under consideration.
Thank you

**Comment by Lakhankumawat** [comment]
I would recommend you to create a new add the required changes and **do not add unnecessary files or lines in README , one c++ file with README only** ,  pr dm me pr number on discord and close this one as there are a lot of changes 

File name must be camel cased with no spacing or underscore in between , just add `FactorialUsingRecursion` and no introduction add necessary changes regarding in README.

**Pull request closed without merging**